packages/eslint-plugin/src/rules/no-redundant-generic-modifiers.ts

```diff
@@ -0,0 +1,91 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+import * as util from '../util';
+
+function isArrowFunctionTypeParameter(node: TSESTree.TSTypeParameter): boolean {
+  return (
+    node.parent?.type === AST_NODE_TYPES.TSTypeParameterDeclaration &&
+    node.parent.parent?.type === AST_NODE_TYPES.ArrowFunctionExpression
```

Comment:
For these `?.`s, the AST types says `node.parent` might not be defined but I don't know of a way to make that happen. Is this an incorrectness in the AST types?

Reply:
Ah, I'll just make this use the same style of ESQuery queries as the old rule.